Automattic/vip-container-images
# ci: fix version extraction regexp

Our current regular expressions for image version extraction for Alpine, nginx, and Traefik have a bug: they extract the hash of the image as well.

For example, the version string we pass to BuildX might look like [this](https://github.com/Automattic/vip-container-images/actions/runs/6612062883/job/17957146189#step:7:9):

```
ghcr.io/automattic/vip-container-images/nginx:1.25.2@sha256:7272a6e0f728e95c8641d219676605f3b9e4759abbdb6b39e5bbd194ce55ebaf
```

That SHA-256 hash must not be passed.

This PR fixes the problem.

To test:
1. Go to https://sed.js.org/
2. Use `-r -e 's/^([^:]+):([^ @$-]+).*/\2/'` as "Command line"
3. Paste
```
FROM alpine:3.18.4@sha256:eece025e432126ce23f223450a0326fbebde39cdf496a85d8c016293fc851978
FROM alpine:3.18.4 AS build
FROM alpine:3.18.4@sha256:eece025e432126ce23f223450a0326fbebde39cdf496a85d8c016293fc851978 AS build
FROM alpine:3.18.4
FROM nginx:1.25.2-alpine@sha256:7272a6e0f728e95c8641d219676605f3b9e4759abbdb6b39e5bbd194ce55ebaf
FROM traefik:2.10.5@sha256:948978f7ec62f137a79f8af7044a1785bd7868706ef2c8cba9c88db688d08661
```

into "STDIN"
4. Expect clean version numbers in "STDOUT | STDERR".
